AppKit/CPView.j

```diff
@@ -3392,6 +3431,153 @@ setBoundsOrigin:
+
+- (void)updateTrackingAreas
+{
+    // You should override this method to remove out of date tracking areas and add recomputed tracking areas;
+    
+    // Cocoa calls this on every view, whereas they have tracking area(s) or not.
+    // Cappuccino behaves differently :
+    // - updateTrackingAreas is called when placing a view in the view hierarchy (that is in a window)
+    // - if you have only CPTrackingInVisibleRect tracking areas attached to a view, it will not be called again (until you move the view in the hierarchy)
+    // - if you have at least one non-CPTrackingInVisibleRect tracking area attached, it will be called every time the view geometry could be modified
+    //   You don't have to touch to CPTrackingInVisibleRect tracking areas, they will be automatically updated
+    //
+    // Please note that it is the owner of a tracking area who is called for updateTrackingAreas.
+    // But, if a view without any tracking area is inserted in the view hierarchy (that is, in a window), the view is called for updateTrackingAreas.
+    // This enables you to use updateTrackingArea to initially attach your tracking areas to the view.
+}
+
+- (void)removeAllTrackingAreas
+{
+    // This utility method is intended for CPView subclasses overwriting updateTrackingAreas
+    //
+    // Typical use would be :
+    //
+    // - (void)updateTrackingAreas
+    // {
+    //      [self removeAllTrackingAreas];
+    //
+    //      ... add your specific updated tracking areas ...
+    //  }
+    
+    while (_trackingAreas.length > 0)
+        [self _removeTrackingArea:_trackingAreas[0]];
+}
+
+// Internal methods
+
+- (void)_removeTrackingArea:(CPTrackingArea)trackingArea
+{
+    if (_window)
+        [_window _removeTrackingArea:trackingArea];
+    
+    [trackingArea setView:nil];
+    
+    [_trackingAreas removeObjectIdenticalTo:trackingArea];
+}
+
+- (void)_notifyUpdateTrackingAreas
+{
+    _inhibitUpdateTrackingAreasNotifications = YES;
+    
+    [self _recursiveNotifyUpdateTrackingAreas];
+    
+    _inhibitUpdateTrackingAreasNotifications = NO;
+}
+
+- (void)_recursiveNotifyUpdateTrackingAreas
+{
+    [self _notifyOwners:[self _prepareOwnersToNotify]];
+    
+    // Recursive call for all subviews
+    
+    for (var i = 0; i < _subviews.length; i++)
+        [_subviews[i] _recursiveNotifyUpdateTrackingAreas];
+}
+
+- (void)_notifyUpdateTrackingAreasForSetWindow
+{
+    // As _setWindow will call recursively itself for subviews, we only have to deal with the view itself
+    // We also do special treatment and notify the view itself if it has no tracking area to
+    // enable the use of updateTrackingAreas as a mean to install tracking areas.
+
+    if ([_trackingAreas count] > 0)
+        var ownersToNotify = [self _prepareOwnersToNotify];
+    else
+        var ownersToNotify = @[ self ];
+    
+    [self _notifyOwners:ownersToNotify]
+}
+
+- (CPArray)_prepareOwnersToNotify
```

Comment:
Then use _calcTrackingAreaOwners. You can't just use _calcOwners; owners of what? It should be clear from the method name.

Yes, _updateTrackingAreasForOwners:(CPArray)owners.